inductiva/mixins/file_manager.py

```diff
@@ -0,0 +1,233 @@
+
+TEMPLATE_EXTENSION = ".jinja"
+SUFFIX_SEPARATOR = "__"
+
+
+class FileManager:
+    """Class for file management."""
+    DEFAULT_ROOT_DIR = "input_dir"
+    __root_dir = None  #pylint: disable=invalid-name
+
+    def __check_root_dir(self):
+        """Set default root folder if unset."""
+        if self.__root_dir is None:
+            self.set_root_dir(self.DEFAULT_ROOT_DIR)
+
+    def set_root_dir(self, root_dir=None):
+        """Set a root directory for the file manager.
+        
+        All files managed through the manager will be
+        placed inside a newly created folder with the given
+        name. If a folder with the same name already exists,
+        the newly created one will be appended with the
+        "__1" suffix. If folders having the "__N" suffix already
+        exist, the given name will be appended with "__{N+1}"
+        where N is determined by the directory with the largest N.
+
+        Args:
+            root_dir: Path to the root directory.
+                If None, an error is raised.
+        """
+
+        if root_dir is None:
+            raise ValueError("Given root directory cannot be None")
+        elif os.path.isdir(root_dir):
+            if format_utils.getenv_bool(
+                    "INDUCTIVA_DISABLE_FILEMANAGER_AUTOSUFFIX", False):
+                raise FileExistsError(f"Directory {root_dir} already exists.")
+            generated_root_dir = _gen_unique_name(root_dir)
+            logging.info(
+                "Directory %s already exists."
+                " Setting root folder to %s.", root_dir, generated_root_dir)
+            root_dir = generated_root_dir
+
+        logging.info("Setting root folder to %s.", root_dir)
+        root_dir = files.resolve_path(root_dir)
+        os.makedirs(root_dir)
+        self.__root_dir = root_dir
+
+    def get_root_dir(self):
+        """Get the active root directory for the file manager."""
+
+        self.__check_root_dir()
+
+        return self.__root_dir
+
+    def add_file(self, source_file, target_file=None, **render_args):
+        """Add a file to the root_dir and render in case it is a template file. 
+        
+        This method has two modes of operation: either it adds `source_file` to
+        the root directory, or it also renders the file (if it has anything to 
+        render) from the given arguments using a templating mechanism. 
+        For the latter case, the file needs to have the templating extension.
+
+        If `target_file` is None, the output is saved inside the manager's root
+        directory with the same name as the source file, but with the
+        template suffix removed.
+        Only template files with the template extension are rendered.
+        Other files are copied.
+
+        Args:
+            source_file: Path to the source file.
+            target_file: Path to the target file.
+            render_args: Arguments to render the template file.
+        """
```

Comment:
We can give a bit more information. Also, a few examples would help.

```suggestion
        """Add a file to the root directory and render it in case it is a template. 
        
        This method copies the contents of the `source_file` to a `target_file`
        inside the root directory. If the source file is a template file, it is rendered
        using the given `render_args` before copying to the target destination.
        If `target_file` is None, the copied file is saved inside the manager's root
        directory with the same name as the source file; if it is a template file, the
        template suffix is removed. 
        A file is considered to be a template file if it ends with the `.jinja` suffix.
        The rendering arguments are ignored if the source file is not a template
        file.

        Args:
            source_file: Path to the source file.
            target_file: Path to the target file.
            render_args: Arguments for rendering the template file.
        
        Examples:
        
        <Add 2 or 3 examples here>
        """
```